pkg/kubelet/dockertools/manager.go

```diff
@@ -213,24 +216,25 @@ func NewDockerManager(
 	}
 
 	dm := &DockerManager{
-		client:                 client,
-		recorder:               recorder,
-		containerRefManager:    containerRefManager,
-		os:                     osInterface,
-		machineInfo:            machineInfo,
-		podInfraContainerImage: podInfraContainerImage,
-		dockerPuller:           newDockerPuller(client, qps, burst),
-		dockerRoot:             dockerRoot,
-		containerLogsDir:       containerLogsDir,
-		networkPlugin:          networkPlugin,
-		livenessManager:        livenessManager,
-		runtimeHelper:          runtimeHelper,
-		execHandler:            execHandler,
-		oomAdjuster:            oomAdjuster,
-		procFs:                 procFs,
-		cpuCFSQuota:            cpuCFSQuota,
-		enableCustomMetrics:    enableCustomMetrics,
-		configureHairpinMode:   hairpinMode,
+		client:                           client,
+		recorder:                         recorder,
+		containerRefManager:              containerRefManager,
+		os:                               osInterface,
+		machineInfo:                      machineInfo,
+		podInfraContainerImage:           podInfraContainerImage,
```

Comment:
If we make secrets as part of ImageSpec, it seems like it would be better to refactor the places where image spec and secrets are passed separately. For example, in func (dm *DockerManager) PullImage(image kubecontainer.ImageSpec, secrets []api.Secret). Are we ok with this refactoring?